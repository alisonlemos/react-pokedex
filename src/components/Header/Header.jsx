import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faRightFromBracket } from '@fortawesome/free-solid-svg-icons';

import { ReactComponent as PokeballLogo } from '../../assets/images/pokeball.svg';
import { ReactComponent as SinviaLogo } from '../../assets/images/synvia-A.svg';
import './Header.scss';
function Header() {
	return (
		<header className="pokedex__header">
			<div className="header__title">
				<PokeballLogo />
				<h1>Pok√©dex</h1>
			</div>
			<div className="header__actions">
				<SinviaLogo />
				<FontAwesomeIcon icon={faRightFromBracket} />
			</div>
		</header>
	);
}

export default Header;
