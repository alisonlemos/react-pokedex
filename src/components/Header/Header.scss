@import '../../assets/colors.scss';

.pokedex__header {
  align-items: center;
  background-color: $red;
  color: $white;
  display: flex;
  justify-content: space-between;
  padding: 0.5rem 10vw;
}

.header__title,
.header__actions {
  align-items: center;
  display: flex;
  gap: 0.5rem;
}
