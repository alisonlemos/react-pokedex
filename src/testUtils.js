import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { orderValues } from './services';
import { setupStore } from './store';

export const mockedPokemonList = {
	data: {
		results: [
			{
				national_number: '99999',
				sprites: {
					normal: 'normal_sprite_url',
				},
				name: 'Pokemon Name',
				type: ['Type_1', 'Type_2'],
			},
		],
	},
};

export const initialState = {
	pokemonList: [],
	favoriteList: [],
	categoriesList: [],
	searchKey: '',
	order: orderValues.NATIONAL_NUMBER_ASC,
	favoriteFilter: false,
	categoriesFilter: [],
	loading: false,
};

export function renderWithProviders(
	ui,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}

	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
