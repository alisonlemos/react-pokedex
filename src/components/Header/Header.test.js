import { renderWithProviders } from '@/testUtils';
import Header from './Header';

describe('Componente Header', () => {
	it('deve renderizar o componente de acordo com o snapshot', () => {
		expect(renderWithProviders(<Header />)).toMatchSnapshot();
	});
});
