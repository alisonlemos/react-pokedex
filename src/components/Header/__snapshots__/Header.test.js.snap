// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Header deve renderizar o componente de acordo com o snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="pokedex__header"
      >
        <div
          class="header__title"
        >
          <svg>
            pokeball.svg
          </svg>
          <h1>
            Pokédex
          </h1>
        </div>
        <div
          class="header__actions"
        >
          <svg>
            synvia-A.svg
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-right-from-bracket "
            data-icon="right-from-bracket"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M160 96c17.7 0 32-14.3 32-32s-14.3-32-32-32H96C43 32 0 75 0 128V384c0 53 43 96 96 96h64c17.7 0 32-14.3 32-32s-14.3-32-32-32H96c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32h64zM504.5 273.4c4.8-4.5 7.5-10.8 7.5-17.4s-2.7-12.9-7.5-17.4l-144-136c-7-6.6-17.2-8.4-26-4.6s-14.5 12.5-14.5 22v72H192c-17.7 0-32 14.3-32 32l0 64c0 17.7 14.3 32 32 32H320v72c0 9.6 5.7 18.2 14.5 22s19 2 26-4.6l144-136z"
              fill="currentColor"
            />
          </svg>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="pokedex__header"
    >
      <div
        class="header__title"
      >
        <svg>
          pokeball.svg
        </svg>
        <h1>
          Pokédex
        </h1>
      </div>
      <div
        class="header__actions"
      >
        <svg>
          synvia-A.svg
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-right-from-bracket "
          data-icon="right-from-bracket"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M160 96c17.7 0 32-14.3 32-32s-14.3-32-32-32H96C43 32 0 75 0 128V384c0 53 43 96 96 96h64c17.7 0 32-14.3 32-32s-14.3-32-32-32H96c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32h64zM504.5 273.4c4.8-4.5 7.5-10.8 7.5-17.4s-2.7-12.9-7.5-17.4l-144-136c-7-6.6-17.2-8.4-26-4.6s-14.5 12.5-14.5 22v72H192c-17.7 0-32 14.3-32 32l0 64c0 17.7 14.3 32 32 32H320v72c0 9.6 5.7 18.2 14.5 22s19 2 26-4.6l144-136z"
            fill="currentColor"
          />
        </svg>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
